### ISSUE TITLE
Qute: with i18n feature in app can not run in native-image mode

### ISSUE THREAD

**Issue by nian8** [quarkusio/quarkus#22962]
### Describe the bug

MessageBundles.DEFAULT_LOCALE is different from MessageBundle.DEFAULT_LOCALE .
so when project with qute message bundle feature can not run in native-image mode. 

### Additional information

```
@MessageBundle(locale = MessageBundles.DEFAULT_LOCALE)
public interface AppMessages {

    /**
     * {msg:hello_name}
     *
     * @param name
     * @return
     */
    @Message("Hello {name ?: \"world\"}!")
    String hello_name(String name);

}
```
when i set locale = MessageBundles.DEFAULT_LOCALE 

i can run the app in native-image mode.

**Comment by mkouba** [comment]
Hm, I cannot see a link between the `MessageBundles.DEFAULT_LOCALE` and `MessageBundle.DEFAULT_LOCALE`. Could you pls provide more information? Ideally a small reproducer app, but at least a more verbose error description...

**Comment by nian8** [comment]
```
public @interface MessageBundle {
    String DEFAULT_LOCALE = "<<default locale>>";
}
```

```
public final class MessageBundles {
   public static final String DEFAULT_LOCALE = "<<default>>";
}
```

later, i will produce a small app, and get more error info.

**Comment by nian8** [comment]
https://github.com/nian8/qute-quickstart

i only follow the https://quarkus.io/guides/qute-reference 3.10 
and 
```java

/**
 *
 */
@MessageBundle
public interface AppMessages {

    /**
     * {msg:hello_name}
     *
     * @param name
     * @return
     */
    @Message("Hello {name ?: \"world\"}!")
    String hello_name(String name);

}
```
so MessageBundle's default locale is "<<default locale>>" . 
when i build native image , app occur an error , 
it can be see in https://github.com/nian8/qute-quickstart/runs/4862077306?check_suite_focus=true 
main info is 
```
java.lang.RuntimeException: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.IllegalStateException: Bean instance for localized interface not found: interface org.acme.qute.message.AppMessages
```

then i skip the test and run the native runner 
the same error occorred
```
2022-01-19 02:22:30,507 ERROR [io.qua.run.Application] (main) Failed to start application (with profile prod): java.lang.IllegalStateException: Bean instance for localized interface not found: interface org.acme.qute.message.AppMessages
	at io.quarkus.qute.i18n.MessageBundles.setupNamespaceResolvers(MessageBundles.java:75)
	at io.quarkus.qute.i18n.MessageBundles_Observer_setupNamespaceResolvers_9feeeffdb84c42f0cd3e7d7e9b8d427d1329f56e.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:320)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:298)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:73)
	at io.quarkus.qute.runtime.EngineProducer.<init>(EngineProducer.java:132)
	at io.quarkus.qute.runtime.EngineProducer_Bean.create(Unknown Source)
	at io.quarkus.qute.runtime.EngineProducer_Bean.create(Unknown Source)
	at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:96)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:29)
	at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:26)
	at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
	at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
	at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:26)
	at io.quarkus.qute.runtime.EngineProducer_Bean.get(Unknown Source)
	at io.quarkus.qute.runtime.EngineProducer_Bean.get(Unknown Source)
	at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:440)
	at io.quarkus.arc.impl.ArcContainerImpl.beanInstanceHandle(ArcContainerImpl.java:453)
	at io.quarkus.arc.impl.ArcContainerImpl.instance(ArcContainerImpl.java:286)
	at io.quarkus.qute.runtime.EngineProducer_Observer_Synthetic_5ca7449e3ab1631e7e8ed910338051c00d29a79b.notify(Unknown Source)
	at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:320)
	at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:302)
	at io.quarkus.arc.impl.EventImpl.fire(EventImpl.java:73)
	at io.quarkus.arc.runtime.ArcRecorder.fireLifecycleEvent(ArcRecorder.java:128)
	at io.quarkus.arc.runtime.ArcRecorder.handleLifecycleEvents(ArcRecorder.java:97)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy_0(Unknown Source)
	at io.quarkus.deployment.steps.LifecycleEventsBuildStep$startupEvent1144526294.deploy(Unknown Source)
	at io.quarkus.runner.ApplicationImpl.doStart(Unknown Source)
	at io.quarkus.runtime.Application.start(Application.java:101)
	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:104)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:67)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
	at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
	at io.quarkus.runner.GeneratedMain.main(Unknown Source)
```

**Comment by mkouba** [comment]
This is really helpful. Thanks @nian8.

**Comment by mkouba** [comment]
Hm, I wasn't able to reproduce the problem with your app.  From the stack trace I can see that the  `MessageBundles.setupNamespaceResolvers()` method fails because Qute is not able to find a bean for the localized interface `org.acme.qute.message.AppMessages` and of the locales (one of the localized message files). I've checked the generated bytecode and it looks ok.

Unfortunately, the error message does not include the locale. I've modified the error messages in this branch: https://github.com/mkouba/quarkus/tree/issue-22962. It would be great if you could try to build quarkus from this branch and test your app against the `999-SNAPSHOT` version.

Anyway, I still don't see a link between `MessageBundle.DEFAULT_LOCALE` and `MessageBundles.DEFAULT_LOCALE`, in other words they should not depend on each other because they're used in different situations.

**Comment by nian8** [comment]
now i reset my app 
```

/**
 * use this is ok
 * {@code @MessageBundle(locale = MessageBundles.DEFAULT_LOCALE)}
 */
@MessageBundle
public interface AppMessages {

    /**
     * {msg:hello_name(name)}
     *
     * @param name
     * @return
     */
    @Message("Hello {name ?: \"world\"}!")
    String hello_name(String name);
}

```
and update my code .
now @mkouba you can build it in native mode , it will be error.

**Comment by nian8** [comment]
i review the qute-i18n source code that MessageBundles.setupNamespaceResolvers() .
in this method, 
![image](https://user-images.githubusercontent.com/13718159/150452543-17c9dc48-4349-40e4-b59e-6fcef8c9e6c8.png)
--- so many code 
i see the condition  
```
// line 64
if (locEntry.getKey().equals(DEFAULT_LOCALE)) {}
```
it is always false when i did not set the MessageBundle.locale

**Comment by mkouba** [comment]
> now i reset my app
> ...
> and update my code .

I did use this version of code and it didn't fail...

> it is always false when i did not set the MessageBundle.locale

Hm, that shouldn't be the case. The `MessageBundles.DEFAULT_LOCALE` key is always used for the default bundle interface [here](https://github.com/quarkusio/quarkus/blob/main/extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java#L274).

What version of Quarkus do yo use? Did you get a chance to try my branch with extended logging?

**Comment by mkouba** [comment]
@nian8 I will close this issue if we don't get more information. Note that the extended logging should be available in quarkus 2.8.

**Comment by mkouba** [comment]
I'm going to close this one as mentioned few months ago. Feel free to reopen if you have more info/questions.
